pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -66,8 +66,22 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 	desiredMinor := getEffectiveMinor(releaseContext.DesiredVersion)
 
 	// if there is no difference in the minor version (4.y.z where 4.y is the same for current and desired), then we can still upgrade
+	// if no cluster overrides have been set
 	if currentMinor == desiredMinor {
 		klog.V(4).Infof("Precondition %q passed: minor from the current %s matches minor from the target %s (both %s).", pf.Name(), cv.Status.History[0].Version, releaseContext.DesiredVersion, currentMinor)
+
+		for _, override := range clusterVersion.Spec.Overrides {
+			if override.Unmanaged {
+				klog.V(4).Infof("ClusterVersionOverridesSet. Z-level upgrade will not start.")
+
+				return &precondition.Error{
+					Nested:  err,
+					Reason:  "ClusterVersionOverridesSet",
+					Message: "Disabling ownership via cluster version overrides prevents upgrades. Please remove overrides before continuing.",
+					Name:    pf.Name(),
```

Comment:
Can we factor [this logic][1] out into a public:

```go
// ClusterVersionOverridesCondition returns an UpgradeableClusterVersionOverrides condition when overrides are set, and nil when no overrides are set.
func ClusterVersionOverridesCondition(cv *configv1.ClusterVersion) *configv1.ClusterOperatorStatusCondition
```

or some such?  And then use that helper here too?  Then future improvements like listing the overridden objects or otherwise copy-editing the reason or message can be made in a single place.

[1]: https://github.com/openshift/cluster-version-operator/blob/cb8241a33846546ab63ec8c01112f633b6980182/pkg/cvo/upgradeable.go#L203-L215